src/lib/utils/member.ts

```diff
@@ -39,117 +39,141 @@ export const getCustomAuthorOptions = async (
+              {
+                role: {
+                  in: getDerivedRoles(
+                    memberPositionIds.map((pos) => pos.id),
+                    true,
+                  ).concat(
+                    memberPositionIds.some((pos) => pos.boardMember)
+                      ? ["dsek.styr"]
+                      : [],
+                  ),
+                },
+              },
+            ],
+          },
+        ],
       },
     })
-  ).concat(member.mandates.map((m) => m.position));
-
-  // Map a doorname to roles, startDate and endDate
-  const allMemberDoors = new Map<
-    string,
-    {
-      roles: string[];
-      startDate: Date | null;
-      endDate: Date | null;
-    }
-  >();
+    .catch(() => {
+      throw error(500, "Could not fetch door access");
+    });
 
-  allDoorPolicies
-    .filter((doorPolicy) =>
-      member.mandates.some(
-        (mandate) =>
-          // A doorpolicy is associated with either a role or a specific member
-          (doorPolicy.role && mandate.positionId.startsWith(doorPolicy.role)) ||
-          doorPolicy.studentId === member.studentId,
-      ),
-    )
-    .forEach((doorPolicy) => {
-      // Get a nice name for a position instead of using the id
-      const positionNamesFromMandates = positions
-        .filter(
-          (position) =>
-            doorPolicy.role && position.id.startsWith(doorPolicy.role),
+  const policiesByDoor: MemberDoorPolicies = userDoorPolicies.reduce(
+    (acc, policy) => {
+      const role = policy.role ?? "Du";
+      const duplicate = acc.find(
+        (p) =>
+          p.name === policy.doorName &&
+          p.startDate === policy.startDatetime &&
+          p.endDate === policy.endDatetime,
+      );
+      if (duplicate) {
+        duplicate.roles.push(role);
+        return acc;
+      }
+      acc.push({
+        name: policy.doorName,
+        roles: [role],
+        startDate: policy.startDatetime,
+        endDate: policy.endDatetime,
+      });
+      return acc;
+    },
+    [] as MemberDoorPolicies,
+  );
+  const memberDoorPolicies: MemberDoorPolicies = policiesByDoor.map(
+    (policy) => {
+      const positionsMappedToThisDoor = memberPositionIds
+        .filter((pos) =>
+          policy.roles.some(
+            (role) =>
+              pos.id.startsWith(role) ||
+              (pos.boardMember && role === "dsek.styr"),
+          ),
         )
-        .map((position) => position.name);
-      const positionNames: string[] =
-        positionNamesFromMandates.length > 0
-          ? positionNamesFromMandates
-          : ["Du"];
-
-      const oldData = allMemberDoors.get(doorPolicy.doorName);
-      const newData = oldData ?? {
-        roles: [...positionNames],
-        startDate: doorPolicy.startDatetime,
-        endDate: doorPolicy.endDatetime,
+        .map((pos) => pos.name);
+      positionsMappedToThisDoor.sort();
+      return {
+        ...policy,
+        roles: positionsMappedToThisDoor || ["Du"],
```

Comment:
This never evaluates to `["Du"]`, maybe something like 
`positionsMappedToThisDoor.length > 0 ? positionsMappedToThisDoor : ["Du"]`